src/ops/mod.rs

```diff
@@ -0,0 +1,32 @@
+use super::runners;
+use super::signatures;
+
+
+/// an an op that updadtes the Runner's data and the signs the runner
+pub trait AbstractOp {
+    type OpSignatureType: signatures::Signature + Clone;
+    type InputDataType: Sized;
+
+    fn add_to<'a: 's, 's>(&'s mut self, mut runner: runners::Runner<'a, Self::InputDataType, Self::OpSignatureType>)
+    -> Result<runners::Runner<'s, runners::RunnerDataBatch<Self::InputDataType, Self::OpSignatureType>, Self::OpSignatureType>, runners::NoMetaDAtaError>
```

Comment:
add todo for clippy warning (no need to fix it for now but evidence that there is a problem)